### ISSUE TITLE
[Roadmap] ms-swift3.6

### ISSUE THREAD

**Issue by Jintao-Huang** [modelscope/ms-swift#4561]
## 模型
1. 最新模型接入 P0
2. Omni pending
    a. 部署支持输出音频
    b. 支持talker的训练
3. All-to-All优化 pending
4. embedding: 支持推理、部署 P0
5. 序列分类: 多标签/回归支持量化

## Megatron-SWIFT 
1. 新模型支持
    a. 多模态: qwen2.5-VL/qwen2.5-Omni P1
    b. DeepSeekV3 P0
2. fp8 P0
3. LoRA支持 P0
4. 支持提前预处理数据集 P1
5. RLHF支持 P0
    a. GRPO
    b. KTO
6. 断点续训 对齐 P0

## 训练
1. RAY支持 P0
2. 长文本ring attention P1

## RL 
1. GRPO
   a. 多轮AsyncEngine P0
   b. Agent MCP
   c. sglang
2. GKD P1 
3. RLOO P1
4. Reinforce++ P1
5. 多模态PPO

### 推理与部署
1. sglang接入推理部署 P0
2. GGUF支持

### 量化
1. bnb支持多模态模型
2. qlora支持merge-lora P1
3. GGUF支持

**Comment by 1571859588** [comment]
希望加入自定义模型forward输入和输出参数，以及处理该参数的loss函数

**Comment by hsushuai** [comment]
deepspeed结合megatron加速训练

**Comment by EvilCalf** [comment]
希望加入一些过程奖励的rl

**Comment by miraiaroha** [comment]
1. 多模态qwen2.5-vl等的Megatron grpo训练支持；
2.  加速算子prefixGrouper；
3. 相同资源（相同节点、gpu数量）下的collocate和异步rollout两种方式的效率对比

**Comment by sosofun** [comment]
1. 期待 DeepSeekV3 / R1
2. 断点续训 对齐

**Comment by shiningliang** [comment]
## 模型
SFT with KD loss. Introduced in [Gemma3 Tech Report](https://arxiv.org/pdf/[2503.19786](https://arxiv.org/pdf/2503.19786)). Seems also used in Qwen3

>  Distillation. We sample 256 logits per token, weighted by teacher probabilities. The student learns the teacher’s distribution with in the samples via cross-entropy loss. The teacher’s target distribution is set to zero probability for non-sampled logits, and renormalized.

## 训练
能否支持DeepSpeed 的AutoTP?  [AutoTP](https://github.com/deepspeedai/DeepSpeed/blob/master/blogs/huggingface-tp/README.md) 给不熟悉Megatron 的同学一个低成本开启并行化的机会

**Comment by vibe-viscot** [comment]
DeepSpeed 的AutoTP，非常想要。同时想要logit知识蒸馏。

**Comment by FoolishMao** [comment]
希望有一个RL的reward合集，各种最佳实践等

**Comment by vibe-viscot** [comment]
希望能让用户可以训练的时候进行数据增强（多模态的，图像、文本、音频的数据增强）

**Comment by RonanDou** [comment]
求一个Megatron-SWIFT的官方镜像

**Comment by xuefei2025** [comment]
deepspeed结合megatron加速训练

**Comment by xuefei2025** [comment]
支持flash attention3加速训练

**Comment by Jintao-Huang** [comment]
> 求一个Megatron-SWIFT的官方镜像

https://swift.readthedocs.io/zh-cn/latest/Instruction/Megatron-SWIFT%E8%AE%AD%E7%BB%83.html

这里有的

**Comment by Jintao-Huang** [comment]
> deepspeed结合megatron加速训练

I checked https://github.com/deepspeedai/Megatron-DeepSpeed, and it seems that the project is not actively maintained anymore. We probably won't support it.

**Comment by sys-reasoner** [comment]
Thanks for the great contribution you have done! Hope this project could support multimodal grpo training (like qwen2.5vl) for megatron.

**Comment by RonanDou** [comment]
我看这里的镜像还是swift3.4.1.post1版本，是需要在这个基础上更新swift到swift3.6嘛？
modelscope-registry.cn-hangzhou.cr.aliyuncs.com/modelscope-repo/modelscope:ubuntu22.04-cuda12.4.0-py311-torch2.6.0-vllm0.8.5.post1-modelscope1.26.0-swift3.4.1.post1

**Comment by CaC033** [comment]
希望支持DAPO

**Comment by vibe-viscot** [comment]
> 希望支持DAPO

不是哥们，DAPO本来就支持的啊，你没在swift用过吗？

**Comment by KooSung** [comment]
希望能支持GME等embedding [MRL](https://arxiv.org/abs/2205.13147) 训练

**Comment by tastelikefeet** [comment]
> 希望能支持GME等embedding [MRL](https://arxiv.org/abs/2205.13147) 训练

是[GME](https://www.modelscope.cn/models/iic/gme-Qwen2-VL-2B-Instruct)还是特指MRL呢

**Comment by KooSung** [comment]
> > 希望能支持GME等embedding [MRL](https://arxiv.org/abs/2205.13147) 训练
> 
> 是[GME](https://www.modelscope.cn/models/iic/gme-Qwen2-VL-2B-Instruct)还是特指MRL呢

@tastelikefeet 我看现在已经支持了GME，希望能支持MRL

**Comment by CaC033** [comment]
希望增加知识蒸馏软标签训练

**Comment by Jintao-Huang** [comment]
> 希望增加知识蒸馏软标签训练

这周支持 稍等

**Comment by AlephZr** [comment]
希望支持SmolVLM

**Comment by Tendo33** [comment]
请问官方的 swift 3.5.0 版本镜像放出来了吗

**Comment by xuefei2025** [comment]
> > deepspeed结合megatron加速训练
> 
> I checked https://github.com/deepspeedai/Megatron-DeepSpeed, and it seems that the project is not actively maintained anymore. We probably won't support it.

可以看下torchtitan这个项目，里面有些加速训练的优化挺好的

**Comment by Jintao-Huang** [comment]
@hsushuai @xuefei2025 
哈咯，支持这个megatron & deepspeed的原因是什么呢，是因为会有更快的速度还是更少的显存

**Comment by Jintao-Huang** [comment]
> 请问官方的 swift 3.5.0 版本镜像放出来了吗

https://github.com/modelscope/ms-swift/pull/4601

**Comment by Summer-Enzhi** [comment]
+1，希望增加知识蒸馏软标签训练，多谢🙏🙏🙏🙏🙏！

**Comment by vibe-viscot** [comment]
请支持一下[Online DPO](https://github.com/huggingface/trl/blob/d9d25a71b2b2c39d11b5bde19e2a9b1fc7315b17/docs/source/online_dpo_trainer.md)

**Comment by Jintao-Huang** [comment]
> +1，希望增加知识蒸馏软标签训练，多谢🙏🙏🙏🙏🙏！

gdk已经支持了

**Comment by groklab** [comment]
request support for other multimodal embeddings.  Thanks.    https://github.com/modelscope/ms-swift/issues/4861

**Comment by HT-NEKO** [comment]
请问reinforce++预计什么时候会支持呢～

**Comment by zxding94** [comment]
请问下grpo的vllm 的 server模式，支持多node训练 同时 单node rollout么

**Comment by hjh0119** [comment]
@zxding94 Sure, set the environment variables NNODES and NODE_RANK in the training script.

**Comment by wanghaichen1** [comment]
megatron 支持 seq_cls

**Comment by zjf0822** [comment]
qwen2.5vl lora微调后，deploy pt vllm lmdeploy 部署，结果不一致的问题希望可以定位解决

**Comment by momo0129** [comment]
qwen2.5-omni-3B/7B 支持 seq_cls

**Comment by Jintao-Huang** [comment]
> qwen2.5vl lora微调后，deploy pt vllm lmdeploy 部署，结果不一致的问题希望可以定位解决

这个问题 应该是vLLM/lmdeploy需要解决的。我们只能保证pt infer_backend与transformers推理对齐。

**Comment by Jintao-Huang** [comment]
> qwen2.5-omni-3B/7B 支持 seq_cls

fixed
https://github.com/modelscope/ms-swift/pull/5329

**Comment by zjf0822** [comment]
许愿，支持Ovis2.5系列模型

**Comment by Jintao-Huang** [comment]
> 许愿，支持Ovis2.5系列模型

https://github.com/modelscope/ms-swift/pull/5426

**Comment by ErenChan** [comment]
希望支持GLM4.5 V的微调

**Comment by manglu097** [comment]
希望支持RLOO，REINFORCE++,REMAX 这些强化学习训练方法

**Comment by Jintao-Huang** [comment]
> 希望支持GLM4.5 V的微调

https://github.com/modelscope/ms-swift/blob/main/examples/megatron/multimodal/moe/glm4_5v.sh

**Comment by zyjcsf** [comment]
请问，支持qwen2.5-omni talker部分的微调了吗

**Comment by JoshonSmith** [comment]
请问是否支持talker的训练了？

**Comment by evelynyhc** [comment]
支持qwen3-omni talker微调了吗
